### PR TITLE
add join proto

### PR DESCRIPTION
**Pull request by hanfei1991**
Signed-off-by: Fei Han <hanfei19910905@gmail.com>

### CODE REVIEW

File: proto/executor.proto [hunk 1]
```diff
@@ -33,13 +34,42 @@ message Executor {
...
+	TiFlash = 2;
 }
 
 message TableScan {
 	optional int64 table_id = 1 [(gogoproto.nullable) = false];
 	repeated ColumnInfo columns = 2;
 	optional bool desc = 3 [(gogoproto.nullable) = false];
 	repeated int64 primary_column_ids = 4;
+	optional EngineType next_read_engine = 5 [(gogoproto.nullable) = false]; // which engine we should in next step, only used by tiflash
+    repeated KeyRange ranges = 6 [(gogoproto.nullable) = false];
```
Comment (SunRunAway) [inline proto/executor.proto:53]:
```suggestion
	repeated KeyRange ranges = 6 [(gogoproto.nullable) = false];
```

Comment (SunRunAway) [reply proto/executor.proto:53]: Could you explain or document it why we need this now but do not need it before?

Comment (hanfei1991) [reply proto/executor.proto:53]: For A Join B, and Table B is in global (broadcast) side. We can know the ranges of A by parsing region info, but cannot get ranges of Table B.

File: proto/executor.proto [hunk 2]
```diff
@@ -33,13 +34,45 @@ message Executor {
...
+	TiFlash = 2;
 }
 
 message TableScan {
 	optional int64 table_id = 1 [(gogoproto.nullable) = false];
 	repeated ColumnInfo columns = 2;
 	optional bool desc = 3 [(gogoproto.nullable) = false];
 	repeated int64 primary_column_ids = 4;
+	optional EngineType next_read_engine = 5 [(gogoproto.nullable) = false]; // which engine we should in next step, only used by tiflash
+	repeated KeyRange ranges = 6 [(gogoproto.nullable) = false];
```
Comment (SunRunAway) [inline proto/executor.proto:53]: Please document this field.

Comment (hanfei1991) [reply proto/executor.proto]: done.

**Pull request merged**